### PR TITLE
MBTiles provider

### PR DESCRIPTION
**Pull request by avbk**
This is the MBTiles Provider including an example.

### CODE REVIEW

**Comment by avbk** [comment]
This is based on #218 and should be merged afterwards

**Comment by alfanhui** [comment]
Probably worth cleaning up your commits. Hope to test this soon as my implementation loading from Sqflite was slow to load. Thanks!

File: lib/flutter_map.dart [hunk 1]
```diff
@@ -14,6 +14,7 @@ import 'package:latlong/latlong.dart';
 export 'src/plugins/plugin.dart';
 export 'src/layer/layer.dart';
 export 'src/layer/tile_layer.dart';
+export 'src/layer/mbitles/mbtiles_image_provider.dart';
```
Comment (cetorres) [inline lib/flutter_map.dart]: There's a typo here:  "mbitles" should be "mbtiles"

Comment (avbk) [reply lib/flutter_map.dart]: Whoopsie, thanks for reviewing it :male_detective:

**Comment by RaimundWege** [comment]
I've tested this PR with an mbtiles (pbf) file I've got from here: https://openmaptiles.com/downloads/tileset/osm/south-america/bolivia/cochabamba/?usage=education

But I get an encoding exception (is it because that pbf isn't supported?):
```
E/flutter (  917): [ERROR:flutter/lib/ui/painting/codec.cc(97)] Failed decoding image. Data is either invalid, or it is encoded using an unsupported format.
I/flutter (  917): ══╡ EXCEPTION CAUGHT BY IMAGE RESOURCE SERVICE ╞════════════════════════════════════════════════════
I/flutter (  917): The following _Exception was thrown resolving an image codec:
I/flutter (  917): Exception: operation failed
I/flutter (  917): 
I/flutter (  917): When the exception was thrown, this was the stack:
I/flutter (  917): #0      MBTileImage._loadAsync (package:flutter_map/src/layer/mbtiles/mbtiles_image_provider.dart:113:14)
I/flutter (  917): <asynchronous suspension>
I/flutter (  917): #1      MBTileImage.load (package:flutter_map/src/layer/mbtiles/mbtiles_image_provider.dart:91:16)
I/flutter (  917): #2      ImageProvider.resolve.<anonymous closure>.<anonymous closure> (package:flutter/src/painting/image_provider.dart:285:105)
I/flutter (  917): #3      ImageCache.putIfAbsent (package:flutter/src/painting/image_cache.dart:157:22)
I/flutter (  917): #4      ImageProvider.resolve.<anonymous closure> (package:flutter/src/painting/image_provider.dart:285:82)
I/flutter (  917): #5      SynchronousFuture.then (package:flutter/src/foundation/synchronous_future.dart:38:29)
I/flutter (  917): #6      ImageProvider.resolve (package:flutter/src/painting/image_provider.dart:283:30)
I/flutter (  917): #7      _ImageProviderResolver.resolve (package:flutter/src/widgets/fade_in_image.dart:333:29)
I/flutter (  917): #8      _FadeInImageState._resolveImage (package:flutter/src/widgets/fade_in_image.dart:408:20)
I/flutter (  917): #9      _FadeInImageState.didChangeDependencies (package:flutter/src/widgets/fade_in_image.dart:390:5)
I/flutter (  917): #10     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3862:12)
I/flutter (  917): #11     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #12     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #13     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #14     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #15     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #16     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #17     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #18     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #19     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #20     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #21     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #22     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #23     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #24     ParentDataElement.mount (package:flutter/src/widgets/framework.dart:4063:11)
I/flutter (  917): #25     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #26     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4982:32)
I/flutter (  917): #27     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #28     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #29     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4876:14)
I/flutter (  917): #30     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #31     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #32     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #33     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #34     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #35     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #36     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #37     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #38     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #39     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #40     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #41     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3864:11)
I/flutter (  917): #42     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #43     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #44     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4982:32)
I/flutter (  917): #45     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #46     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #47     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4876:14)
I/flutter (  917): #48     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #49     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #50     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #51     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #52     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #53     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #54     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #55     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #56     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4876:14)
I/flutter (  917): #57     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #58     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #59     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4876:14)
I/flutter (  917): #60     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #61     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #62     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #63     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #64     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #65     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3864:11)
I/flutter (  917): #66     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #67     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #68     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #69     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #70     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #71     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #72     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #73     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #74     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #75     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4876:14)
I/flutter (  917): #76     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #77     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #78     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4876:14)
I/flutter (  917): #79     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #80     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #81     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #82     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #83     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #84     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3864:11)
I/flutter (  917): #85     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #86     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #87     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #88     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #89     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #90     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #91     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #92     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #93     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #94     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (  917): #95     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (  917): #96     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (  917): #97     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3864:11)
I/flutter (  917): #98     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (  917): #99     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (  917): #100    Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (  917): #101    _LayoutBuilderElement._layout.<anonymous closure> (package:flutter/src/widgets/layout_builder.dart:118:18)
I/flutter (  917): #102    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2253:19)
I/flutter (  917): #103    _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:107:11)
I/flutter (  917): #104    RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:1728:58)
I/flutter (  917): #105    PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:797:15)
I/flutter (  917): #106    RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:1728:13)
I/flutter (  917): #107    _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:205:5)
I/flutter (  917): #108    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #109    RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:798:17)
I/flutter (  917): #110    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #111    RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:199:11)
I/flutter (  917): #112    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #113    MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:142:11)
I/flutter (  917): #114    _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:350:7)
I/flutter (  917): #115    MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:212:7)
I/flutter (  917): #116    RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:356:14)
I/flutter (  917): #117    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #118    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #119    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #120    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #121    _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1206:11)
I/flutter (  917): #122    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #123    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #124    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #125    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #126    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #127    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #128    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #129    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #130    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #131    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #132    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #133    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
E/flutter (  917): [ERROR:flutter/lib/ui/painting/codec.cc(97)] Failed decoding image. Data is either invalid, or it is encoded using an unsupported format.
I/chatty  (  917): uid=10047(com.johnpryan.leafletflutterexample) 1.io identical 3 lines
E/flutter (  917): [ERROR:flutter/lib/ui/painting/codec.cc(97)] Failed decoding image. Data is either invalid, or it is encoded using an unsupported format.
I/flutter (  917): #134    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #135    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #136    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #137    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #138    RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:3032:13)
I/flutter (  917): #139    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #140    RenderStack.performLayout (package:flutter/src/rendering/stack.dart:510:15)
I/flutter (  917): #141    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #142    __RenderTheatre&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #143    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #144    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #145    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #146    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #147    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #148    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #149    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #150    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #151    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #152    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
I/flutter (  917): #153    RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
I/flutter (  917): #154    RenderView.performLayout (package:flutter/src/rendering/view.dart:151:13)
I/flutter (  917): #155    RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1507:7)
I/flutter (  917): #156    PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:766:18)
I/flutter (  917): #157    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:329:19)
I/flutter (  917): #158    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:701:13)
I/flutter (  917): #159    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:268:5)
I/flutter (  917): #160    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:988:15)
I/flutter (  917): #161    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:928:9)
I/flutter (  917): #162    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:749:7)
I/flutter (  917): #171    _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
I/flutter (  917): #172    _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
I/flutter (  917): #173    _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
I/flutter (  917): (elided 8 frames from package dart:async)
I/flutter (  917): 
I/flutter (  917): Image provider: MBTileImage()
I/flutter (  917): Image key: MBTileImage()
I/flutter (  917): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by avbk** [comment]
Hi Raimund, I'm sorry, but this tile provider is only intended to read MBTIles files.

As to the documentation in OSM wiki PBF is a custom binary format (https://wiki.openstreetmap.org/wiki/PBF_Format) while MBTiles basically is a SQLite database (https://wiki.openstreetmap.org/wiki/MBTiles). So the wording on the webpage you mentioned seems wrong, a file cannot be PBF and MBTiles at the same time.

**Comment by emallay** [comment]
To further clarify: PBF is a binary / compressed representation of raw OSM data (which is normally just pure text). There are no imagery tiles in the PBF. The raw OSM data (text or PBF) is used as the source content when creating the tiles. Those tiles are then put into an MBTiles database (SQLite), which is what flutter_map reads. 

If you want a (relatively) easy, free way to create MBTiles databases, take a look at Maperitive.  A more powerful alternative is Mapnik (which is what OpenStreetMap uses itself for tiles) but it comes with a much steeper learning curve.

**Comment by avbk** [comment]
@emallay thanks for the clarification :+1:

**Comment by mdoelker** [comment]
@avbk @emallay Actually PBF in this case refers to the tile format. From the [spec](https://github.com/mapbox/mbtiles-spec/blob/master/1.3/spec.md#content):

> pbf as a format refers to gzip-compressed vector tile data in [Mapbox Vector Tile](https://github.com/mapbox/vector-tile-spec/) format.

That said, this format is obviously not supported here yet.

**Pull request merged**